bundles/pubsub/pubsub_spi/include/pubsub_protocol.h

```diff
@@ -28,39 +28,77 @@
 
 typedef struct pubsub_protocol_header pubsub_protocol_header_t;
 
+/**
+ * The protocol header structure, contains the information about the message payload and metadata
+ */
 struct pubsub_protocol_header {
-    unsigned int msgId;
-    unsigned short msgMajorVersion;
-    unsigned short msgMinorVersion;
-
-    unsigned int payloadSize;
-    unsigned int metadataSize;
+  /** message payload identification attributes */
+    unsigned int msgId; /*!< Message id of the payload */
+    unsigned short msgMajorVersion; /*!< Message Major version of the payload */
+    unsigned short msgMinorVersion; /*!< Message Minor version of the payload */
+
+    /** Payload and metadata sizes attributes */
+    unsigned int payloadSize; /*!< Size of the payload (Can be used for payload buffer allocation by the receiver) */
+    unsigned int metadataSize; /*!< Size of the metadata (Can be used for metadata buffer allocation by the receiver) */
+
+    /** message segmentation attributes */
```

Comment:
If I read the code correct the segmentation attributes are only written if segmentation is supported. 
Could the documentation reflect this, so this it is clear that this change does not break the curren wire protocol (and still plays nice with its Java counterpart)

Reply:
I have updated the documentation